runtime/frequency/src/lib.rs

```diff
@@ -150,6 +148,31 @@ impl Contains<RuntimeCall> for BaseCallFilter {
 	}
 }
 
+impl BaseCallFilter {
+	fn is_utility_call_allowed(call: &RuntimeCall) -> bool {
+		match call {
+			RuntimeCall::Utility(pallet_utility::Call::batch { calls, .. }) |
+			RuntimeCall::Utility(pallet_utility::Call::batch_all { calls, .. }) |
+			RuntimeCall::Utility(pallet_utility::Call::force_batch { calls, .. }) =>
+				!calls.iter().any(Self::is_blocked_call),
+			_ => false,
+		}
+	}
+
+	fn is_blocked_call(call: &RuntimeCall) -> bool {
+		match call {
```

Comment:
I suggest calling this `is_pays_no_call`.

What would happen if we did not allow any free transaction to be called using the Utility batch?

```
if call.get_dispatch_info().pays_fee == Pays::No {
   true
}

false
```

Reply:
If that works ^ I like it better. That way the blocked call list doesn't require maintenance

Reply:
@wilwade  would we want to block every `Pays::No` call? I dont think so, then we will end up blocking basic system calls too, like remark etc.
@enddynayn what you think

Reply:
We are blocking these only when calling batch.

Reply:
@saraswatpuneet I think we do want to. What pays::no call might we want to allow?

Reply:
@enddynayn Ah yes. You are correct. We only want to block them from batches. So I think we do want to block all from batching.